src/pages/permissions/panels/GroupOrIdentityChangesTable.tsx

```diff
@@ -0,0 +1,159 @@
+  onChangeGroupBy: () => void;
+  authUserName: string;
+  identityGroupsChangeSummary: IdentitiesOrGroupsChangeSummary;
+  groupIdentitiesChangeSummary: IdentitiesOrGroupsChangeSummary;
+}
+
+const generateRowsFromIdentityGroupChanges = (
+  identityGroupsChangeSummary: IdentitiesOrGroupsChangeSummary,
+  authUserName: string,
+) => {
+  const identityIDs = Object.keys(identityGroupsChangeSummary);
+
+  const rows: JSX.Element[] = [];
+  for (const id of identityIDs) {
+    const groupChangesForIdentity = identityGroupsChangeSummary[id];
+    const identityLoggedIn = id === authUserName;
+    let isFirstRow = true;
+    Array.from(groupChangesForIdentity.added).forEach((group, idx) => {
+      const rowKey = `${id}-${group}-added-${idx}`;
+      const row = (
+        <tr className={isFirstRow ? "first-row" : ""} key={rowKey}>
+          <td>
+            {isFirstRow ? (
+              <>
+                {groupChangesForIdentity.name}{" "}
+                <Tag show={identityLoggedIn}>You</Tag>
+              </>
+            ) : (
+              ""
+            )}
+          </td>
+          <td>+ {group}</td>
+        </tr>
+      );
+      rows.push(row);
+      isFirstRow = false;
+    });
+
+    Array.from(groupChangesForIdentity.removed).forEach((group, idx) => {
+      const rowKey = `${id}-${group}-removed-${idx}`;
+      const row = (
+        <tr className={isFirstRow ? "first-row" : ""} key={rowKey}>
+          <td>
+            {isFirstRow ? (
+              <>
+                {groupChangesForIdentity.name}{" "}
+                <Tag show={identityLoggedIn}>You</Tag>
+              </>
+            ) : (
+              ""
+            )}
+          </td>
+          <td className="remove-group">- {group}</td>
+        </tr>
+      );
+      rows.push(row);
+      isFirstRow = false;
+    });
+  }
+
+  return rows;
+};
+
+const generateRowsFromGroupIdentityChanges = (
+  groupIdentitiesChangeSummary: IdentitiesOrGroupsChangeSummary,
+  authUserName: string,
+) => {
+  const groups = Object.keys(groupIdentitiesChangeSummary);
+
+  const rows: JSX.Element[] = [];
+  for (const group of groups) {
+    const identityChangesForGroup = groupIdentitiesChangeSummary[group];
+    let isFirstRow = true;
+    Array.from(identityChangesForGroup.added).forEach((identity, idx) => {
+      const rowKey = `${group}-${identity}-added-${idx}`;
+      const identityLoggedIn = identity === authUserName;
+      const row = (
+        <tr className={isFirstRow ? "first-row" : ""} key={rowKey}>
+          <td>{isFirstRow ? identityChangesForGroup.name : ""}</td>
+          <td>
+            + {identity} <Tag show={identityLoggedIn}>You</Tag>
+          </td>
+        </tr>
+      );
+      rows.push(row);
+      isFirstRow = false;
+    });
+
+    Array.from(identityChangesForGroup.removed).forEach((identity, idx) => {
+      const rowKey = `${group}-${identity}-removed-${idx}`;
+      const identityLoggedIn = identity === authUserName;
```

Comment:
Instead of identity, the left-hand side of the comparison needs to be the IDs, or this will fail if the name doesn't match the ID.

Also, `authUserName` being the ID of an identity is a bit confusing.

Reply:
changed `authUserName` to `loggedInIdentityID`. The root cause of the issue here was in `pivotIdentityGroupsChangeSummary`, instead of adding ids I was adding names, this is now fixed.